community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -309,23 +308,23 @@ private void loadIdGenerator()
     protected void verifyCorrectTypeDescriptorAndVersion() throws IOException
     {
         String expectedTypeDescriptorAndVersion = getTypeAndVersionDescriptor();
-        int length = UTF8.encode( expectedTypeDescriptorAndVersion ).length;
-        byte bytes[] = new byte[length];
-        ByteBuffer buffer = ByteBuffer.wrap( bytes );
-        long fileSize = getFileChannel().size();
-        if ( fileSize >= length )
+        try ( PagedFile pagedFile = pageCache.map( storageFileName, pageCache.pageSize() ) )
         {
-            getFileChannel().position( fileSize - length );
+            readTypeDescriptorAndVersion = StoreVersionCheck.readVersion( pagedFile, expectedTypeDescriptorAndVersion );
         }
-        else
+        if ( readTypeDescriptorAndVersion == null )
         {
             setStoreNotOk( new IllegalStateException(
-                    "Invalid file size " + fileSize + " for " + this + ". Expected " + length + " or bigger" ) );
+                    "No trailer present in store, expected " + expectedTypeDescriptorAndVersion ) );
+            return;
+        }
+        if ( readTypeDescriptorAndVersion.trim().length() < expectedTypeDescriptorAndVersion.length() )
```

Comment:
Ideally, `StoreVersionCheck.readVersion` would return a value that does not need trimming.